yazi-core/src/tab/commands/copy.rs

```diff
@@ -43,3 +66,114 @@ impl Tab {
 		futures::executor::block_on(CLIPBOARD.set(s));
 	}
 }
+
+#[derive(Default, Clone, Copy)]
+enum PathSeparator {
+	Unix,
+	#[default]
+	Auto,
+}
+
+impl From<&Cmd> for PathSeparator {
+	fn from(c: &Cmd) -> Self {
+		match c.str("separator") {
+			Some("unix") => PathSeparator::Unix,
+			Some("auto") => PathSeparator::Auto,
+			_ => Default::default(),
+		}
+	}
+}
+
+#[cfg(not(target_os = "windows"))]
+fn path_to_os_str(path: &Path, _separator: PathSeparator) -> Cow<'_, OsStr> {
+	return Cow::Borrowed(path.as_os_str());
+}
+
+#[cfg(target_os = "windows")]
+fn path_to_os_str(path: &Path, separator: PathSeparator) -> Cow<'_, OsStr> {
+	if let PathSeparator::Auto = separator {
+		return Cow::Borrowed(path.as_os_str());
+	};
+
+	use std::path::Component;
+
+	let mut s = String::with_capacity(path.as_os_str().len());
+	for component in path.components() {
+		match component {
+			Component::RootDir => {}
+			Component::CurDir => s.push('.'),
+			Component::ParentDir => s.push_str(".."),
+			Component::Normal(path) => s.push_str(&path.to_string_lossy()),
+			Component::Prefix(prefix) => {
+				// "C:\foo" => [Prefix("C:"), RootDir, Normal(foo)]
+				s.push_str(&prefix.as_os_str().to_string_lossy());
+				// If we push a "/" below, we will met a RootDir and push a "/"
+				// again resulting in "C://". So we need to skip that.
+				continue;
+			}
+		};
+		s.push('/');
+	}
+
+	use std::os::windows::ffi::OsStrExt as _;
+	use std::path::MAIN_SEPARATOR;
+
+	let path_ends_with_main_sep =
+		path.as_os_str().encode_wide().last() == Some(MAIN_SEPARATOR as u16);
+	if !path_ends_with_main_sep && s != "/" && s.ends_with('/') {
+		s.pop();
+	}
```

Comment:
Remove the trailing slash if necessary to prevent files from being identified as directories and to align with the input format.